datafusion/sqllogictest/test_files/array.slt

```diff
@@ -6232,12 +6244,12 @@ select array_intersect(arrow_cast([1, 1, 2, 2, 3, 3], 'LargeList(Int64)'), null)
 query ?
 select array_intersect(null, [1, 1, 2, 2, 3, 3]);
 ----
-NULL
+[]
```

Comment:
is it possible to keep null?

Reply:
Above we have
```
query ?
select array_intersect([1, 1, 2, 2, 3, 3], null);
----
[]
```

So I think it's better that `array_intersect` is commutative. I can make both `null` or both empty list but I think it should be the same if we flip the arguments. Which one should it be?

Reply:
Also, below we have:
```
query ?
select array_intersect([], null);
----
[]
```